SQLQuery_Week09_Homework.sql

```diff
@@ -0,0 +1,414 @@
+INSERT INTO Librarians
+VALUES ('Catalin Popescu','2009-07-08', 1);
+INSERT INTO Librarians
+VALUES ('Stoian Petru','2015-02-04', 0);
+INSERT INTO Librarians
+VALUES ('Cristian Ioan','2012-08-09', 1);
+INSERT INTO Librarians
+VALUES ('Mircea Popa','2010-11-20', 0);
+INSERT INTO Librarians
+VALUES ('Carol Dumitru','2011-09-08', 0);
+INSERT INTO Librarians
+VALUES ('Robert Dumitrescu','2019-10-10', 1);
+
+INSERT INTO Sales
+VALUES(22,1,2);
+INSERT INTO Sales
+VALUES(64,2,1);
+INSERT INTO Sales
+VALUES(25,3,3);
+INSERT INTO Sales
+VALUES(72,4,4);
+INSERT INTO Sales
+VALUES(14,5,5);
+INSERT INTO Sales
+VALUES(34,6,6);
+INSERT INTO Sales
+VALUES(41,1,7);
+
+INSERT INTO Books_Libraries
+VALUES (1,5);
+INSERT INTO Books_Libraries
+VALUES (2,3);
+INSERT INTO Books_Libraries
+VALUES (3,4);
+INSERT INTO Books_Libraries
+VALUES (4,2);
+INSERT INTO Books_Libraries
+VALUES (5,1);
+INSERT INTO Books_Libraries
+VALUES (6,4);
+INSERT INTO Books_Libraries
+VALUES (7,6);
+INSERT INTO Books_Libraries
+VALUES (8,2);
+INSERT INTO Books_Libraries
+VALUES (9,3);
+INSERT INTO Books_Libraries
+VALUES (10,2);
+INSERT INTO Books_Libraries
+VALUES (5,3);
+INSERT INTO Books_Libraries
+VALUES (5,1);
+INSERT INTO Books_Libraries
+VALUES (2,5);
+INSERT INTO Books_Libraries
+VALUES (1,3);
+INSERT INTO Books_Libraries
+VALUES (7,1);
+INSERT INTO Books_Libraries
+VALUES (8,2);
+INSERT INTO Books_Libraries
+VALUES (10,3);
+
+INSERT INTO Books_Sales
+VALUES (1,1);
+INSERT INTO Books_Sales
+VALUES (3,2);
+INSERT INTO Books_Sales
+VALUES (5,3);
+INSERT INTO Books_Sales
+VALUES (6,4);
+INSERT INTO Books_Sales
+VALUES (7,5);
+INSERT INTO Books_Sales
+VALUES (9,6);
+INSERT INTO Books_Sales
+VALUES (10,7);
+
+-- 2. Display all the librarians together with the library name they work in ordered by librarian name and library name descending.
+
+SELECT l.Name, LI.Name
+FROM Sales s
+    INNER JOIN Librarians l ON l.Id = s.LibrarianId
+    INNER JOIN Libraries LI ON LI.Id = s.LibraryId
+ORDER BY l.Name, LI.Name DESC;
+
+-- 3. Update all the street names to start with "STR.".
+
+UPDATE Libraries
+SET StreetName = CONCAT('STR. ', StreetName);
+
+-- 4. Display the authors' email for those who have more than 3 books published.
+
+SELECT a.Email 'Authors''s email with more than 3 books published'
+FROM Books b
+    INNER JOIN Authors a ON a.Id = b.AuthorId
+GROUP BY a.Email
+HAVING COUNT(AuthorId) > 3;
+
+-- 5. Display the library name and the book title of the largest sale (the highest number of copies sold in one sale).
+
+GO
+CREATE VIEW BookIdAndNumberOfCopiesByLibrary
+AS
+    SELECT b.Title, s.LibraryId, s.NumberOfCopies
+    FROM Books_Sales bs
+     INNER JOIN Books b ON b.Id = bs.BookId
+     INNER JOIN Sales s ON s.Id = bs.SalesId;
+GO
+
+SELECT SUM(bv.NumberOfCopies), l.Name
+FROM BookIdAndNumberOfCopiesByLibrary bv
+    INNER JOIN Libraries l ON l.Id = bv.LibraryId
+GROUP BY l.Name
+HAVING SUM(bv.NumberOfCopies) = (SELECT MAX(NumberOfCopies) FROM BookIdAndNumberOfCopiesByLibrary)
```

Comment:
You overcomplicated things a bit with the view (it's not required) but for practicing it it's great.